### PR TITLE
Add workflow for deployment

### PR DESCRIPTION
**Pull request by arjunchib**

### CODE REVIEW

File: ecosystem.config.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "apps":
+    [
+      {
+        "script": "./src/index.js",
+        "name": "memebot",
+        "env": { "NODE_ENV": "production", "COMMAND_PREFIX": "!mb"},
```
Comment (MatthewHannah) [inline ecosystem.config.json]: Do you want the command prefix to be set only here? I saw you changed the example to !ml but I'm not sure if .env overrides this.

File: ecosystem.config.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+        "max_memory_restart": "900M",
+        "exp_backoff_restart_delay": 100,
+        "max_restarts": 10
```
Comment (MatthewHannah) [inline ecosystem.config.json]: Do we really need a max restarts? It keeps memebot alive and I wasn't having crash issues with it.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Add secrets to PM2 Config
+        env:
+          MEMEBOT_API_BASE_URL: ${{ secrets.MEMEBOT_API_BASE_URL }}
+          DISCORD_TOKEN: ${{ secrets.DISCORD_TOKEN }}
+        run: node ${GITHUB_WORKSPACE}/scripts/pm2-config.js DISCORD_TOKEN MEMEBOT_API_BASE_URL
+      - name: Upload files to memebot
+        run: rsync -a --delete --exclude ".git" -e "ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i ${HOME}/memebot_rsa" ${GITHUB_WORKSPACE}/ ${USERNAME}@${HOST}:~/memebot/
+      - name: Reload PM2 server
+        run: |
+          ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i ${HOME}/memebot_rsa ${USERNAME}@${HOST} -T << EOF
```
Comment (MatthewHannah) [inline .github/workflows/main.yml]: Is it worth it in the future to try to actually add the host key into the secrets or some other sort of config var for the server so we don't need to work around that check?

File: ecosystem.config.json (hunk 1 above)
Comment (arjunchib) [reply ecosystem.config.json]: yeah I think I forgot to update, i'll make that `!ml`

File: ecosystem.config.json (hunk 2 above)
Comment (arjunchib) [reply ecosystem.config.json]: I did have some crashes, but we can probably just use the exponential backoffs and that should be enough.

File: .github/workflows/main.yml (hunk 3 above)
Comment (arjunchib) [reply .github/workflows/main.yml]: Yeah we can probably just add the server to the known hosts file?

**Comment by MatthewHannah** [comment]
Did you want to update the !ml or just merge?

**Comment by arjunchib** [comment]
I will update

File: .github/workflows/main.yml (hunk 3 above)
Comment (arjunchib) [reply .github/workflows/main.yml]: Updated this should be more secure now

**Comment by arjunchib** [comment]
Ah I did not understand that well. I had forgot about the ecosystem file prefix, but it was correct, I wanted `!mb` to be for production and `!ml` to be for local. I usually do `.env-example` for local so its easier for people to get setup.

**Pull request merged**